language/compiler/ir_to_bytecode/src/context.rs

```diff
@@ -289,6 +299,93 @@ impl<'a> Context<'a> {
         Ok(())
     }
 
+    //**********************************************************************************************
+    // Source Maps
+    //**********************************************************************************************
+
+    pub fn add_top_level_function_mapping(
```

Comment:
Can you just make the source map a public field? Then you don't need all these delegates

Reply:
Or you can make the map not part of the context, whatever is easier. But I'd prefer no delegates :)